### ISSUE TITLE
Allow the filter prop of ReferenceArrayField to handle arrays

### ISSUE THREAD

**Issue by srosset81** [marmelab/react-admin#5871]
**Is your feature request related to a problem? Please describe.**
The `ReferenceArrayField` component has a `filter` prop which allows to filter results client-side. However if the filtered value is an array, the filter does not work because the component does a simple `==` comparison.
For example if I have a resource with a property `foo: ['bar', 'baz']`, then using the filter prop `{ foo: 'bar' }` will not return this property because `'bar' != ['bar', 'baz']`.

**Describe the solution you'd like**
Change [this line](https://github.com/marmelab/react-admin/blob/master/packages/ra-core/src/controller/field/useReferenceArrayFieldController.ts#L177) in the `useReferenceArrayFieldController` to something like this:

```js
const recordValue = get(record, filterName);
return Array.isArray(recordValue) ? recordValue.includes(filterValue) : filterValue == recordValue;
```

**Describe alternatives you've considered**
Create our own component, but I think this is a feature others may want.

**Additional context**
I can do a PR if you think this is a good idea.

**Comment by djhi** [comment]
Hi and thanks for the report. Would you mind creating a PR for this?

**Comment by srosset81** [comment]
Done here: https://github.com/marmelab/react-admin/pull/5887

**Comment by fzaninotto** [comment]
Fixed by #5887
